runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingPipelineTranslator.java

```diff
@@ -99,6 +99,13 @@ public CompositeBehavior enterCompositeTransform(TransformHierarchy.Node node) {
 
     PTransform<?, ?> transform = node.getTransform();
     if (transform != null) {
+      // TODO(BEAM-10670): Remove this and the ReadTranslator once the "use_deprecated_read"
+      // experiment is removed. Don't translate composite Read transforms since we expect the
+      // primitive expansion containing an SDF to be used.
+      if (PTransformTranslation.READ_TRANSFORM_URN.equals(
+          PTransformTranslation.urnForTransformOrNull(transform))) {
+        return CompositeBehavior.ENTER_TRANSFORM;
+      }
```

Comment:
How will this work with the opt-out of SDF? Will primitive Reads still be translated then?

Reply:
The non SDF version of Read is always a primitive transform so it gets replaced down below in `visitPrimitiveTransform`.

Reply:
Makes sense!